app/routers/api/identity.py

```
@@ -0,0 +1,98 @@
+                status_code=status.HTTP_409_CONFLICT,
+                detail=f"This action will delete {existing_connections_count} existing remote connections. "
+                "Please confirm by setting confirm_replace=True.",
+            )
+
+    try:
+        identity_service.import_keys_pem(db, request.signing_private_key, request.kx_private_key)
+    except ValueError as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e),
+        )
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)